src/archivematicaCommon/lib/namespaces.py

```diff
@@ -45,6 +47,7 @@ def xml_find_premis(elem, path):
     """``find`` with PREMIS 2 fallback."""
     matches = elem.find(path, namespaces=NSMAP)
     if matches is None:
+        ElementPath._cache.clear()
```

Comment:
I'd appreciate a note explaining why is this needed. Maybe at the module level, as it's done in almost all the functions.